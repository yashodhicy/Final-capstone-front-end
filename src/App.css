body {
  min-height: 100vh;
  min-width: 100vw;
}

#root {
  width: 100%;
  height: 100%;
  display: flex;
  position: relative;
}

#pages-content {
  width: 100%;
  overflow-y: auto;
}

@media (min-width: 768px) {
  #root {
    height: auto;
    margin: 0;
    padding: 0;
  }

  #pages-content {
    display: flex;
    margin-left: 23%;
  }
}
