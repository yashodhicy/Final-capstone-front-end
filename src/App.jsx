import { Route, Routes } from 'react-router-dom'
import Navbar from './components/Navbar'
import Houses from './components/House'
import HouseDetails from './HouseDetails'
import { getHouses } from './Redux/HouseSlice';
import { useDispatch } from 'react-redux';
import { useEffect } from 'react';

function App() {

  const dispatch = useDispatch();
  
  useEffect(() => {
    dispatch(getHouses());
  }, [dispatch]);

  return (
      <>
        {/* <Navbar /> */}
        <Routes>
          <Route path='/' element= {<Houses />}></Route>
          <Route path='/houses/:houseId' element = { <HouseDetails />}>
          </Route>
        </Routes>
        
        
        
    </>
  )
}

export default App
