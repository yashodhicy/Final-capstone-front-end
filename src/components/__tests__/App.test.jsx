import renderer from 'react-test-renderer';
import { render } from "@testing-library/react";
import { createBrowserRouter, createRoutesFromElements, Route, RouterProvider } from "react-router-dom";
import App from "../../App";

describe("App", () => {
  test("renders without errors", () => {
    const tree = renderer
    const router = createBrowserRouter(
      createRoutesFromElements(
        <Route
          errorElement={<div>This is the error page</div>}
          path="/"
          element={<div>MainLayout</div>}
        >
          <Route path="/" element={<div>Root</div>}>
            {/* Add navigation tabs here */}
            <Route path="/" element={<div>Houses</div>}></Route>
            <Route path="/add" element={<div>AddHouse</div>} />
            <Route path="/houses/:houseId" element={<div>HouseDetails</div>} />
            <Route path="/delete" element={<div>DeleteHouse</div>} />
            <Route path="/reserve" element={<div>Reservation</div>} />
          </Route>
          <Route path="/login" element={<div>Signin</div>} />
          <Route path="/signup" element={<div>Signup</div>} />
        </Route>
      )
    );

    render(<RouterProvider router={router}><App /></RouterProvider>);
    // Add your assertions here
    expect(tree).toMatchSnapshot();
  });
});