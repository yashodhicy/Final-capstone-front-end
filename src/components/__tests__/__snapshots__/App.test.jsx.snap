// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without errors 1`] = `
{
  "_Scheduler": {
    "reset": [Function],
    "unstable_IdlePriority": 5,
    "unstable_ImmediatePriority": 1,
    "unstable_LowPriority": 4,
    "unstable_NormalPriority": 3,
    "unstable_Profiling": null,
    "unstable_UserBlockingPriority": 2,
    "unstable_advanceTime": [Function],
    "unstable_cancelCallback": [Function],
    "unstable_clearYields": [Function],
    "unstable_continueExecution": [Function],
    "unstable_flushAll": [Function],
    "unstable_flushAllWithoutAsserting": [Function],
    "unstable_flushExpired": [Function],
    "unstable_flushNumberOfYields": [Function],
    "unstable_flushUntilNextPaint": [Function],
    "unstable_forceFrameRate": [Function],
    "unstable_getCurrentPriorityLevel": [Function],
    "unstable_getFirstCallbackNode": [Function],
    "unstable_next": [Function],
    "unstable_now": [Function],
    "unstable_pauseExecution": [Function],
    "unstable_requestPaint": [Function],
    "unstable_runWithPriority": [Function],
    "unstable_scheduleCallback": [Function],
    "unstable_setDisableYieldValue": [Function],
    "unstable_shouldYield": [Function],
    "unstable_wrapCallback": [Function],
    "unstable_yieldValue": [Function],
  },
  "act": [Function],
  "create": [Function],
  "unstable_batchedUpdates": [Function],
}
`;
